src/mbedtls_context.cpp

```diff
@@ -585,7 +587,38 @@ namespace sockpp {
     {
         if (!root_cert_locator_)
             return -1;
-        string certData((const char*)child->raw.p, child->raw.len);
+        
+        // Construct the cert chain including all intermediates in PEM format:
+        string certData;
+        for (auto crt = child; crt; crt = crt->next) {
+            int ret = 0;
+            size_t olen = 10000; // initial buffer size
+            std::vector<unsigned char> buf;
+            for (int i = 0; i < 2; i++) {
+                buf.resize(olen);
+                ret = mbedtls_pem_write_buffer("-----BEGIN CERTIFICATE-----\n",
+                                               "-----END CERTIFICATE-----\n",
+                                               crt->raw.p, crt->raw.len,
+                                               buf.data(), buf.size(), &olen);
+                if (ret != MBEDTLS_ERR_BASE64_BUFFER_TOO_SMALL) {
```

Comment:
Is this correct?  It looks like it only runs the loop twice if the buffer isn't big enough??

Reply:
NVM.... I get it.  `mbedtls_pem_write_buffer` will set olen to the size that it needs, if it does not have enough room?

Reply:
Yes, the olen will get updated with the min buffer to use.